src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -777,5 +777,16 @@ function discoverController(
     return loadedIndexPattern;
   }
 
+  // Block the UI from loading if the user has loaded a rollup index pattern but it isn't
+  // supported.
+  $scope.isUnsupportedRollup = (
```

Comment:
when rollup reference is abstracted away, maybe this can be something like: 

```js
import {
    hasSearchStategyForIndexPattern,
    isIndexPatternSupported,
    renderUnsupportedIndexPatternMessage,
} from 'ui/courier';

if(
    !hasSearchStategyForIndexPattern($route.current.locals.ip.loaded)
    || !isIndexPatternSupported($route.current.locals.ip.loaded)
) {
   // psuedo code - implementation will need adjustment for rendering logic! :)
    $scope.unsupportedMessage = renderUnsupportedIndexPatternMessage();
    return;
}

Reply:
I redact my previous review after chatting with CJ and realizing that the unsupported rollup messages were are due to giving user hints when they're using OSS. X-Pack code isn't reachable in that scenario.

The comprise to support this type of error in OSS, while avoiding cross stream OSS/X-Pack code and rollup references, is to check whether the index pattern is the default type, instead of checking for rollup type.